app/scripts/modules/amazon/instance/details/instance.details.controller.js

```diff
@@ -361,7 +363,10 @@ module.exports = angular.module('spinnaker.instance.detail.aws.controller', [
       );
     };
 
-    retrieveInstance().then(() => app.registerAutoRefreshHandler(retrieveInstance, $scope));
+    retrieveInstance().then(() => {
+       let refreshWatcher = app.autoRefreshStream.subscribe(retrieveInstance);
+       $scope.$on('$destroy', () => refreshWatcher.dispose());
+     });
```

Comment:
This is what I'm proposing we do, but there are more than 50 methods following this pattern throughout the codebase, so I want to get buy-in before applying the pattern everywhere.